app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala

```diff
@@ -113,19 +107,43 @@ class AuditController @Inject() (
     if (request.headers.get(Constants.XAuditMetaPath).isEmpty) {
       EitherT.leftT[Future, Details](PresentationError.badRequestError(s"${Constants.XAuditMetaPath} is missing"))
     } else {
-      val metadata = Metadata(request.headers)
+      val metadataRequest = MetadataRequest(request.headers)
       payload match {
         case Left(summary) =>
           val objSummary = Json.obj(
             "objectSummary"    -> summary.objectSummary.toString,
             "additionalFields" -> summary.fields.toString()
           )
-          EitherT.rightT[Future, PresentationError](Details(metadata, Some(objSummary)))
+          EitherT.rightT[Future, PresentationError](
+            Details(
+              None,
+              Metadata(
+                metadataRequest.path,
+                metadataRequest.movementId,
+                metadataRequest.messageId,
+                metadataRequest.enrolmentEORI,
+                metadataRequest.movementType,
+                metadataRequest.messageType
+              ),
+              Some(objSummary)
+            )
+          )
         case Right(s) =>
           for {
             body <- extractBody(s)
             src  <- parse[JsObject](body)
-          } yield Details(metadata, Some(src))
+          } yield Details(
+            None,
+            Metadata(
+              metadataRequest.path,
+              metadataRequest.movementId,
+              metadataRequest.messageId,
+              metadataRequest.enrolmentEORI,
+              metadataRequest.movementType,
+              metadataRequest.messageType
+            ),
+            Some(src)
+          )
       }
     }
```

Comment:
This is weird because in the same method, you're controlling the creation of `MetadataRequest`, followed by turning it into a `Metadata` immediately.

Step back and think about this -- does this make sense? Probably not. The idea of `MetadataRequest` and `DetailsRequest` was important to avoid overposting -- a problem when you're using an ORM like play-json, but it isn't a problem with the headers because you're handling the parsing yourself. In this scenario (extracting data from headers), I'd argue that you should build a `Details` object directly here because you don't need to protect against overposting.

I get that I said to make the `DetailsRequest` and `MetadataRequest` objects, that was important in the case of status events because you're using the model to parse the Json. I should have made that clearer. Here, I'd probably not build a `MetadataRequest` object and just create a `Metadata` object directly from the headers, but importantly **not by creating an `apply` method on the `Metadata` object**.